addons/web/static/tests/helpers/mock_server.js

```diff
@@ -94,6 +103,26 @@ var MockServer = Class.extend({
         }
         return fvg;
     },
+    /**
+     * Simulates a complete fetch call.
+     *
+     * @param {string} resource
+     * @param {Object} init
+     * @returns {any}
+     */
+    async performFetch(resource, init) {
+        if (this.debug) {
+            console.log(
+                '%c[fetch] request ' + resource, 'color: blue; font-weight: bold;',
+                JSON.parse(JSON.stringify(init))
+            );
+        }
+        const res = await this._performFetch(resource, init);
+        if (this.debug) {
+            console.log('%c[fetch] response' + resource, 'color: blue; font-weight: bold;', res);
+        }
+        return res;
+    },
```

Comment:
Any name suggestion?

I think this fetch mock should actually replace the "fake mock rpc" that are used for example for img src, etc.

Reply:
Maybe just add "browser" in the name should avoid confusion, like `performBrowserFetch()`.
Also I'm not found of having a public and private variant of a method with same name. It seems you do this just to decorate before/after in case of debug mode, but I think it's best to dump all in a single (public) method `performBrowserFetch()` altogether.

Reply:
A single method does not allow to plug override in the correct place, this is the same for rpc.

Reply:
You don't need to override private method right now, so that's no motivation to do that. I think no method should have the same name, regardless of visibility.

Reply:
`_performFetch` is overridden in mail mock server.

Reply:
Ok, my bad I didn't see that.

I quite dislike the split of method in public/private variants. This is motivated by how we extend behaviour (patching by override), which is an aspect of OOP I have become against (in short: structure of code becomes extremely weak to changes).

This is (web) mock server code, and in some ways "legacy", so ok to make a pass for this pattern here.